src/pages/Manager.vue

```diff
@@ -499,30 +499,47 @@ import CategoryFilterModal from '../components/modals/CategoryFilterModal.vue';
 		}
 
         async setAllModsEnabled(enabled: boolean) {
-            for (const mod of this.localModList) {
-                let profileErr: R2Error | void;
-                if (enabled) {
-                    profileErr = await ProfileInstallerProvider.instance.enableMod(mod, this.contextProfile!);
-                } else {
-                    profileErr = await ProfileInstallerProvider.instance.disableMod(mod, this.contextProfile!);
-                }
-                if (profileErr instanceof R2Error) {
-                    this.showError(profileErr);
-                    continue;
-                }
-                const update: ManifestV2[] | R2Error = await ProfileModList.updateMod(mod, this.contextProfile!, async (updatingMod: ManifestV2) => {
+            let lastSuccessfulUpdate: ManifestV2[] = [];
+
+            try {
+                for (const mod of this.localModList) {
+                    if (mod.isEnabled() === enabled) {
+                        continue;
+                    }
+
+                    let profileErr: R2Error | void;
                     if (enabled) {
-                        updatingMod.enable();
+                        profileErr = await ProfileInstallerProvider.instance.enableMod(mod, this.contextProfile!);
                     } else {
-                        updatingMod.disable();
+                        profileErr = await ProfileInstallerProvider.instance.disableMod(mod, this.contextProfile!);
                     }
-                });
-                if (update instanceof R2Error) {
-                    this.showError(update);
-                    continue;
+                    if (profileErr instanceof R2Error) {
+                        this.showError(profileErr);
+                        continue;
+                    }
+                    const update: ManifestV2[] | R2Error = await ProfileModList.updateMod(mod, this.contextProfile!, async (updatingMod: ManifestV2) => {
+                        if (enabled) {
+                            updatingMod.enable();
+                        } else {
+                            updatingMod.disable();
+                        }
+                    });
+                    if (update instanceof R2Error) {
+                        this.showError(update);
+                    } else {
+                        lastSuccessfulUpdate = update;
+                    }
+                }
+            } catch (e) {
+                const name = `Error ${enabled ? "enabling" : "disabling"} mods`;
+                const message = (e as Error).message;
+                this.showError(new R2Error(name, message));
```

Comment:
There's a `R2Error.fromThrownValue` class method (or something like that) available which you can use for this kind of stuff btw

Reply:
Changed to use `fromThrownValue`.